daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/DependencyVersion.scala

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.validation
+
+import com.daml.lf.data.Ref._
+import com.daml.lf.language.Ast._
+import com.daml.lf.transaction.VersionTimeline
+
+private[validation] object DependencyVersion {
+
+  @throws[ValidationError]
+  def checkModule(world: World, pkgId: PackageId, mod: Module): Unit = {
+    import VersionTimeline.Implicits._
+
+    val modLangVersion = mod.languageVersion
+    val modName = mod.name
+    val ctx = ContextModule(pkgId, modName)
+
+    val dependencies = world.idsInModule(ctx, pkgId, modName).collect {
+      case Identifier(depPkgId, QualifiedName(depModName, _))
```

Comment:
Why do we care about the module name here? The LF version is per package and hence getting all the referenced package IDs should be enough, isn't it?

Reply:
Looking at the Scala AST, it does have the version on the module not the package. No idea why, that’s definitely not how we represent things in protobuf or on the Haskell side.

Maybe let’s keep it like this for this PR and then fix that separately?